lib/galaxy/datatypes/binary.py

```diff
@@ -198,6 +198,18 @@ def display_peek(self, dataset):
             return dataset.peek
         except Exception:
             return "Compressed binary file (%s)" % (nice_size(dataset.get_size()))
+    
+    def sniff(self, filename):
+        # If the zip file contains multiple files then return true
+        zf = zipfile.ZipFile(filename)
+        zf_files = zf.infolist()
+        # Filter the directories within the file list
+        zf_files = [x.filename for x in zf_files if x.file_size > 0]
+        # Filter hidden file within the file list
+        hidden_file_patterns = ["__MACOSX",".DS_Store","^\."]
+        regex = re.compile('|'.join(hidden_file_patterns))
+        zf_files = [x for x in zf_files if x not in list(filter(regex.search, zf_files))]
+        return len(zf_files) > 1
```

Comment:
```suggestion
        UNDESIRED_FILES = {'__MACOSX', '.DS_Store'}
        with zipfile.ZipFile(filename) as zf:
            zf_files = zf.infolist()
            count = 0
            for f in zf_files:
                if x.file_size > 0 and not f.filename in UNDESIRED_FILES and not f.filename.startswith('.'):
                    count += 1
                if count > 1:
                    return True
```
That should have the same effect, but close the ZipFile after we're done and avoids the regex.

Reply:
Hum ...
regexp were efficiant for this:
```
dossier1/.DS_Store
__MACOSX/dossier1/._.DS_Store
```

My new proposal:
```python
    def sniff2(filename):
        UNDESIRED_FILES = {'.*__MACOSX.*', '.*\.DS_Store.*'}
        regex = re.compile('|'.join(UNDESIRED_FILES))
        with zipfile.ZipFile(filename) as zf:
            zf_files = zf.infolist()
            count = 0
            for f in zf_files:
                if f.file_size > 0 and not regex.match(f.filename) and not f.filename.startswith('.'):
                    count += 1
                if count > 1:
                    return True
```

Reply:
Why do you want to use a regex without a regular expression? Simple set lookup is going to be a lot more efficient.

Reply:
`        UNDESIRED_FILES = {'.*__MACOSX.*', '.*\.DS_Store.*'}`
It was a misstake, now there are some real regexp 😛